src/map/map.cpp

```diff
@@ -325,15 +325,14 @@ void gamemap::overlay(const gamemap& m, const config& rules_cfg, int xpos, int y
 			}
 			else {
 				set_terrain(map_location(x2,y2), t);
-				if (false /*TODO: add overwrite special locs paramter*/) {
-					starting_positions_.right.erase(t_translation::coordinate(x2, y2));
-					for (auto& pair : m.starting_positions_.right.equal_range(t_translation::coordinate(x1, y1))) {
-						starting_positions_.insert(tstarting_positions::value_type(pair.second, t_translation::coordinate(x2, y2)));
-					}
-				}
 			}
 		}
 	}
+
+	for(auto& pair : m.starting_positions_.left) {
+		starting_positions_.left.erase(pair.first);
+		starting_positions_.insert(tstarting_positions::value_type(pair.first, t_translation::coordinate(pair.second.x + xpos, pair.second.y + ypos)));
```

Comment:
I think its not as eay as just adding a fixed offset to pair.second, due to the hex-based map, the code in the above loop above also does a more complicated check in the y coordinate to make sure it also wokrs with odd x positions.

Also, you might want to wrap the loob body ina  if(on_map(target loc)) check.